### PR TITLE
refactor(MenuItem): separate basic logic from MenuItem to BaseMenuItem

### PR DESCRIPTION
**Pull request by YossiSaadi**
https://monday.monday.com/boards/3532715121/pulses/6600529809

### CODE REVIEW

File: packages/core/src/components/Menu/MenuItem/components/BaseMenuItem/BaseMenuItem.tsx [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+              disabled={disabled}
+            />
+            <MenuItemSubMenu
+              anchorRef={anchorRef}
+              open={shouldShowSubMenu}
+              onClose={closeSubMenu}
+              autoFocusOnMount={!useDocumentEventListeners}
+            >
+              {subMenu}
+            </MenuItemSubMenu>
```
Comment (talkor) [inline packages/core/src/components/Menu/MenuItem/components/BaseMenuItem/BaseMenuItem.tsx:156]: Why are these in the Base and not in the MenuItem?

File: packages/core/src/components/Menu/MenuItem/components/BaseMenuItem/BaseMenuItem.types.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { MenuItemProps } from "../../MenuItem";
+import { MenuChild } from "../../../Menu/MenuConstants";
+import React from "react";
+import { VibeComponentProps } from "../../../../../types";
+
+export type LossMenuItemProps = Omit<
```
Comment (talkor) [inline packages/core/src/components/Menu/MenuItem/components/BaseMenuItem/BaseMenuItem.types.ts:6]: What's this for?

File: packages/core/src/components/Menu/MenuItem/components/BaseMenuItem/BaseMenuItem.tsx (hunk 1 above)
Comment (YossiSaadi) [reply packages/core/src/components/Menu/MenuItem/components/BaseMenuItem/BaseMenuItem.tsx:156]: cause I want to be able to add sub menu for CustomMenuItem and for each MenuItem variant
MenuItem is an opinionated implemenation now, but the basic should have the capabilities for sub menu

File: packages/core/src/components/Menu/MenuItem/components/BaseMenuItem/BaseMenuItem.types.ts (hunk 2 above)
Comment (YossiSaadi) [reply packages/core/src/components/Menu/MenuItem/components/BaseMenuItem/BaseMenuItem.types.ts:6]: to not cause breaking change, I'm taking the MenuItem props, but just what I need, and to override those who are conflicting

File: packages/core/src/components/Menu/MenuItem/components/BaseMenuItem/BaseMenuItem.tsx (hunk 1 above)
Comment (talkor) [reply packages/core/src/components/Menu/MenuItem/components/BaseMenuItem/BaseMenuItem.tsx:156]: Is this intended to replace the option to pass a component to the menu item using the title prop?

Comment (YossiSaadi) [reply packages/core/src/components/Menu/MenuItem/components/BaseMenuItem/BaseMenuItem.tsx:156]: CustomMenuItem is the one intended to replace, basically yes.
but it puts more weight on the developer to "know what he's doing".
BaseMenuItem is internal

Comment (talkor) [reply packages/core/src/components/Menu/MenuItem/components/BaseMenuItem/BaseMenuItem.tsx:156]: Sounds legit to me, but I'm wondering why should devs use the old MenuItem currently and not just the new CustomMenuItem? It gives them more flexibility in the first place and behaves the same

Comment (YossiSaadi) [reply packages/core/src/components/Menu/MenuItem/components/BaseMenuItem/BaseMenuItem.tsx:156]: they shouldn't basically
only for tooltip reasons (if they want)

**Pull request merged**